src/components/SearchBarPopup/SearchBarPopup.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { RefObject } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import Modal from "../../components/Modal/Modal";
+import "./searchBarPopup.scss";
+import _ from "lodash";
+
+interface Destination {
+  id: number;
+  name: string;
+  address: string;
+  isAdded: boolean;
+}
+
+/**
+ * Search bar states
+ * searchKey: Keyword to perform search with
+ * results: Search results retrieved from backend call
+ * loading: True if a search is being performed, false if no search currently performed
+ * errMsg: Any error message from search
+ */
+export interface SearchBarState {
+  searchKey: string;
+  results: Array<Destination>;
+  resultsOpen: boolean;
+  loading: boolean;
+  errMsg: string;
+}
+
+/**
+ * A popup to search for places and to be added into a Trecipe
+ */
+export class SearchBarPopup extends React.Component<{}, SearchBarState> {
+  private container: RefObject<HTMLDivElement> = React.createRef<
```

Comment:
This ref is never assigned to any element in `render()`

Reply:
Probably another leftover from copy pasting SearchBar component LOL